youtube_dl/extractor/nova.py

```diff
@@ -43,6 +43,10 @@ def _real_extract(self, url):
 
         formats = []
         for format_id, format_list in bitrates.items():
+            if format_id == 'hls':
+                formats.extend(self._extract_m3u8_formats(
+                    format_list, video_id, ext='mp4', m3u8_id='hls', fatal=False))
```

Comment:
You must validate `format_list`. Read coding conventions.

Reply:
This I didn't figure out -- I don't quite understand what you mean, could you please be more specific?

Reply:
`format_list` is not guaranteed to be an URL.

Reply:
Thanks and apologies, resolved.